Make ONNXProgram.model_proto and disk file the same

thiagocrepaldi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #122230
* __->__ #122196

Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

Commit:
Make ONNXProgram.model_proto and disk file the same

Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

[ghstack-poisoned]

BowenBao commented on torch/onnx/_internal/fx/serialization.py:
```diff
@@ -22,7 +22,7 @@ def _create_tensor_proto_with_external_data(
     location: str,
     basepath: str,
-    dtype_override: Optional["onnx.TypeProto"] = None,
+    dtype_override: Optional["onnx.TypeProto"] = None,  # type: ignore[name-defined]
```
this needs to be in the first PR

Commit:
Update on "Make ONNXProgram.model_proto and disk file the same"


Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

[ghstack-poisoned]

Commit:
Update on "Make ONNXProgram.model_proto and disk file the same"


Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

[ghstack-poisoned]

thiagocrepaldi commented:
@pytorchbot merge

Commit:
Update on "Make ONNXProgram.model_proto and disk file the same"


Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

[ghstack-poisoned]

Commit:
Update on "Make ONNXProgram.model_proto and disk file the same"


Currently, the in-memory onnx program model proto does
not contain initializers saved into the disk version.

This PR changes this behavior, so that both versions are
identical. This is important for running models with fake
tensor from OMMProgram.model_proto directly, without a file

[ghstack-poisoned]

thiagocrepaldi commented:
@pytorchbot merge

thiagocrepaldi commented:
@pytorchbot merge -f "Flaky merge. All CI is green, but pytorchbot says 4 jobs have failed"
